specification/http/1.0/openapi.json

```diff
@@ -0,0 +1,456 @@
+                      "description": "JSON Schema describing the output parameters for the tool. Supports standard JSON Schema validation but excludes $ref and definitions/schemas for simplicity."
+                    }
+                  }
+                }
+              },
+              "required": ["available_modes"],
+              "allOf": [
+                {
+                  "if": {
+                    "properties": {
+                      "available_modes": {
+                        "contains": { "const": "value" }
+                      }
+                    }
+                  },
+                  "then": {
+                    "required": ["value"]
+                  }
+                }
+              ],
+              "additionalProperties": false
+            },
+            "requirements": {
+              "type": "object",
+              "properties": {
+                "authorization": {
+                  "type": "array",
+                  "items": {
+                    "type": "object",
+                    "properties": {
+                      "id": {
+                        "type": "string",
+                        "description": "A provider's unique identifier, allowing the tool to specify a specific authorization provider."
+                      },
+                      "oauth2": {
+                        "type": "object",
+                        "description": "OAuth 2.0-specific authorization details.",
+                        "properties": {
+                          "scopes": {
+                            "type": "array",
+                            "items": {
+                              "type": "string"
+                            }
+                          }
+                        },
+                        "additionalProperties": false
+                      }
+                    }
+                  },
+                  "required": ["id"],
+                  "additionalProperties": false
+                },
+                "secrets": {
+                  "type": "array",
+                  "items": {
+                    "type": "object",
+                    "properties": {
+                      "id": {
+                        "type": "string",
+                        "description": "The secret's unique identifier."
+                      }
+                    },
+                    "required": ["id"],
+                    "additionalProperties": false
+                  }
+                }
+              },
+              "additionalProperties": false
+            }
+          },
+          "required": ["id", "name", "description", "toolkit"],
+          "additionalProperties": false
+        },
+        "ExecuteToolRequest": {
+          "type": "object",
+          "properties": {
+            "execution_id": {
+              "type": "string",
+              "description": "The globally-unique ID for this tool execution."
```

Comment:
* Is there any recommendation for how to assign a unique ID? 
* What is the scope of "global" (restricted to given tenant probably?)

Reply:
Clarified that this can be something like a UUID, but does not have to be truly global.